staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc_test.go

```diff
@@ -306,7 +311,19 @@ func (c *claimsTest) run(t *testing.T) {
 		t.Fatalf("serialize token: %v", err)
 	}
 
-	got, ok, err := a.AuthenticateToken(testContext(t), token)
+	ctx := testContext(t)
+	// wait for the authenticator to be initialized
+	err = wait.PollUntilContextCancel(ctx, time.Millisecond, true, func(context.Context) (bool, error) {
+		if v, _ := a.idTokenVerifier(); v == nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("failed to initialize the authenticator: %v", err)
+	}
```

Comment:
Uh what changed to need this?

Reply:
With the `"discovery-url"` unit test, I'm trying to initialize the provider by fetching keys from the claims server (`fetchKeysFromRemote`) and skipping configuring a static key set [here](https://github.com/kubernetes/kubernetes/pull/123527/files#diff-9dfc52cf08b2714b4e54cfc514504fbf637c9182648855e7bc7a3f210d6ea63dR273-R276). With this for the test, the provider is initialized async and there is some delay. This is to ensure the provider is ready before calling `AuthenticateToken`.

I had this wrapped in a `if c.fetchKeysFromRemote` but figured this check shouldn't have any impact on the static key set cases either.